style/tests.qmd

```diff
@@ -0,0 +1,68 @@
+---
+title: "Test style guide"
+description: "Guidelines on how to write tests of our Python code"
+---
+
+{{< include /includes/_wip.qmd >}}
+
+## What to write tests for
+
+Write to write tests for:
+
+- Every function, class, and method.
+- Every possible input and output.
+- Every possible error within the function.
+
+## How to write tests
+
+### Test location
+
+We use [pytest](https://docs.pytest.org/en/stable/) for testing our
+Python code and keep the [tests outside the application
+code](https://docs.pytest.org/en/stable/explanation/goodpractices.html#tests-outside-application-code)
+in a `tests` directory, like so:
+
+``` plaintext
+sprout/
+├── core/
+│   ├── my_function.py
+├── tests/
+│   ├── test_my_function.py
+```
+
+### Test files
+
+Make one test file per Python script and only test the functions within
+that script. This makes it easier to do a cycle of writing and testing,
+since you only need to run the one test file and not all the tests.
+
+Always name the test file the same as the Python script, but with
+`test_` as a prefix. E.g., `my_function.py` --\> `test_my_function.py`.
+
+### Test names
+
+Use the following general pattern for naming tests:
+
+- `test_<action>_<condition>`. E.g.,
```

Comment:
This is not taken from any framework or anything I have read on how to name tests, but rather just what it seems like we're (mostly) doing right now

Reply:
I think this was inspired by Philip...